components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -0,0 +1,242 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.nimbus
+
+import android.content.Context
+import android.content.pm.PackageInfo
+import android.content.pm.PackageManager
+import android.os.Build
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import mozilla.components.service.glean.Glean
+import mozilla.components.support.base.coroutines.Dispatchers
+import mozilla.components.support.base.log.Log
+import uniffi.nimbus.AppContext
+import uniffi.nimbus.EnrolledExperiment
+import uniffi.nimbus.ExperimentConfig
+import uniffi.nimbus.NimbusClient
+import java.io.File
+import java.util.Locale
+
+/**
+ * This is the main experiments API, which is exposed through the global [Nimbus] object.
+ */
+open class NimbusInternalAPI internal constructor() {
+    companion object {
+        private const val LOG_TAG = "service/Nimbus"
+        private const val EXPERIMENT_COLLECTION_NAME = "nimbus-mobile-experiments"
+        internal const val NIMBUS_DATA_DIR: String = "nimbus_data"
+
+        const val KINTO_ENDPOINT_DEV = "https://kinto.dev.mozaws.net"
+        const val KINTO_ENDPOINT_STAGING = "https://settings.stage.mozaws.net"
+        const val KINTO_ENDPOINT_PROD = "https://firefox.settings.services.mozilla.com"
```

Comment:
Is this safe from a security standpoint? These values couldn't be swapped after compiling, right?

Reply:
We have some precedence for this need from `FxaAccountManager` where it accepts a data class configuration `ServerConfig` so that the app that set this up based on build variants, non-Firefox builds, etc.

You can see the usage of this style [here in Fenix](https://github.com/mozilla-mobile/fenix/blob/db2cee4a129fc2a15413be84bf9c71faa1aa5c73/app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt#L75-L89).